test/rules/overridden_fields.dart

```diff
@@ -0,0 +1,130 @@
+// Copyright (c) 2021, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(OverriddenFieldsTest);
+  });
+}
+
+@reflectiveTest
+class OverriddenFieldsTest extends LintRuleTest {
+  @override
+  String get lintRule => 'overridden_fields';
+
+  /// https://github.com/dart-lang/linter/issues/2874
+  test_conflictingStaticAndInstance() async {
+    await assertNoLint(r'''
+class A {
+  static final String field = 'value';
+}
+
+class B extends A {
+  String field = 'otherValue';
+}
+''');
+  }
+
+  test_recursiveInterfaceInheritance() async {
+    // Produces a recursive_interface_inheritance diagnostic.
+    await assertNoLint(r'''
+class A extends B {}
+class B extends A {
+  int field = 0;
+}
+''');
+  }
+
+  test_mixinInheritsFromNotObject() async {
```

Comment:
I would strongly encourage you to have a separate test method for each lint diagnostic. It's much easier to understand what's being tested.

Reply:
Agreed.  This test source is not trivial to disentangle and I'm not confident that a lot of it has lasting value so I decided to leave it as is w/ the TODO below.

That said, in general, I'm with you 100%.

Thanks!